lib/ex_aws/request/url.ex

```diff
@@ -14,7 +14,7 @@ defmodule ExAws.Request.Url do
     |> convert_port_to_integer
     |> (&struct(URI, &1)).()
     |> URI.to_string()
-    |> String.trim_trailing("?")
+    |> String.replace_suffix("?", "")
```

Comment:
This is needed to support `text.txt?` object, since if the object name ends with the question mark, the `URI.to_string` will add another question mark and the `String.trim_trailing/2` will remove both.